src/main/java/frc/robot/subsystems/VisionSubsystem.java

```diff
@@ -175,122 +113,114 @@ public VisionSubsystem(DrivebaseWrapper aprilTagsHelper) {
             .getEntry();
   }
 
-  // method called to update results that cameras detect
   public void update() {
-    for (PhotonPipelineResult result : leftCamera.getAllUnreadResults()) {
-      process(result, photonPoseEstimatorLeftCamera, rawFieldPose3dEntryLeft);
+
+    RawFiducial[] rawFiducialsL = leftCamera.getRawFiducials();
+    if (rawFiducialsL != null) {
+      for (RawFiducial rf : rawFiducialsL) {
+        processLimelight(leftCamera, rawFieldPose3dEntryLeft, rf);
+      }
     }
-    for (PhotonPipelineResult result : rightCamera.getAllUnreadResults()) {
-      process(result, photonPoseEstimatorRightCamera, rawFieldPose3dEntryRight);
+    RawFiducial[] rawFiducialsR = rightCamera.getRawFiducials();
+    if (rawFiducialsR != null) {
+
+      for (RawFiducial rf : rawFiducialsR) {
+        processLimelight(rightCamera, rawFieldPose3dEntryRight, rf);
+      }
     }
   }
 
-  // processs current result with cameras and camera pose estimator and where swerve believes it is
-  private void process(
-      PhotonPipelineResult result,
-      PhotonPoseEstimator estimator,
-      StructPublisher<Pose3d> rawFieldPose3dEntry) {
-    // makes a different timestamp to keep track of time
-    var RawTimestampSeconds = result.getTimestampSeconds();
-    // for waiting until orange pi 5 syncing and getting rid old results
-    if (!MathUtil.isNear(Timer.getFPGATimestamp(), RawTimestampSeconds, 5.0)) {
-      return;
-    }
-    // updates estimated pose
-    var estimatedPose = estimator.update(result);
-    // if there is an actual pose then it gets a timestamp and pose3d
-    if (estimatedPose.isPresent()) {
-      var TimestampSeconds = estimatedPose.get().timestampSeconds;
-      var FieldPose3d = estimatedPose.get().estimatedPose;
-      // sets full pose3d for logging
-      rawFieldPose3dEntry.set(FieldPose3d);
-      // if you disabled vision then vision doesn't do anything anymore
-      if (disableVision.getBoolean(false)) {
+  private void processLimelight(
+      LLCamera camera, StructPublisher<Pose3d> rawFieldPoseEntry, RawFiducial rf) {
+    if (disableVision.getBoolean(false)) return;
+    PoseEstimate estimate = camera.getPoseEstimate();
+
+    if (estimate != null) {
+      if (estimate.tagCount == 0) {
         return;
       }
-      //////// below is if a tag is a specific tag id don't read this result
-      // if (BadAprilTagDetector(result)) {
-      //   return;
```

Comment:
This is now just calling `LimelightHelpers.SetFiducialIDFiltersOverride` sometime earlier (once the alliance is confidently known) with a list of the tag IDs to include for localization (everything else is ignored).